src/main/java/uk/gov/companieshouse/psc/delta/processor/PscDeltaProcessor.java

```diff
@@ -2,36 +2,64 @@
 
 import static java.lang.String.format;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.messaging.Message;
-import org.springframework.messaging.MessageHeaders;
 import org.springframework.stereotype.Component;
 import uk.gov.companieshouse.api.delta.PscDelta;
+import uk.gov.companieshouse.api.psc.FullRecordCompanyPSCApi;
+import uk.gov.companieshouse.delta.ChsDelta;
 import uk.gov.companieshouse.logging.Logger;
-import uk.gov.companieshouse.psc.delta.service.api.ApiClientServiceImpl;
-
+import uk.gov.companieshouse.psc.delta.exception.NonRetryableErrorException;
+import uk.gov.companieshouse.psc.delta.service.api.ApiClientService;
+import uk.gov.companieshouse.psc.delta.transformer.PscApiTransformer;
 
 @Component
 public class PscDeltaProcessor {
 
+    private final PscApiTransformer transformer;
     private final Logger logger;
-    private final ApiClientServiceImpl apiClientService;
+    private final ApiClientService apiClientService;
 
     @Autowired
-    public PscDeltaProcessor(Logger logger, ApiClientServiceImpl apiClientService) {
+    public PscDeltaProcessor(Logger logger, ApiClientService apiClientService, PscApiTransformer transformer) {
         this.logger = logger;
         this.apiClientService = apiClientService;
+        this.transformer = transformer;
     }
 
     /**
      * Process PSC Statement Delta message.
      */
-    public void processDelta(Message<PscDelta> pscDelta) {
-        final MessageHeaders headers = pscDelta.getHeaders();
-        final PscDelta payload = pscDelta.getPayload();
-        logger.info(format("Successfully extracted PSC Delta of %s", payload.toString()));
+    public void processDelta(Message<ChsDelta> chsDelta) {
+        final ChsDelta payload = chsDelta.getPayload();
+        final String contextId = payload.getContextId();
+
+        FullRecordCompanyPSCApi fullRecordCompanyPscApi = new FullRecordCompanyPSCApi();
```

Comment:
This should be on line 53 and you shouldn't be instantiating it, just declaring.